pkg/broker/ensure.go

```diff
@@ -204,6 +205,18 @@ func CreateNewBrokerRoleBinding(kubeClient kubernetes.Interface, serviceAccount,
 
 // nolint:wrapcheck // No need to wrap here
 func CreateNewBrokerSA(kubeClient kubernetes.Interface, submarinerBrokerSA, inNamespace string) (brokerSA *v1.ServiceAccount, err error) {
-	return kubeClient.CoreV1().ServiceAccounts(inNamespace).Create(
-		context.TODO(), NewBrokerSA(submarinerBrokerSA), metav1.CreateOptions{})
+	sa := NewBrokerSA(submarinerBrokerSA)
+	_, err = serviceaccount.Ensure(kubeClient, inNamespace, sa, true)
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = serviceaccount.EnsureSecretFromSA(kubeClient, sa.Name, inNamespace)
```

Comment:
My previous comment still applies, unless I’ve missed something:

> I would prefer the secret handling to be hidden in the SA handling — callers shouldn’t have to know that to set up a service account correctly, they have to set up a secret.

In other words, there should only be a call to `serviceaccount.Ensure()` in this function, and `serviceaccount.Ensure()` should handle the secret (perhaps by calling `EnsureSecretFromSA()`, of course, but that should be transparent to the caller).

Reply:
> My previous comment still applies, unless I’ve missed something:
> 
> > I would prefer the secret handling to be hidden in the SA handling — callers shouldn’t have to know that to set up a service account correctly, they have to set up a secret.
> 
> In other words, there should only be a call to `serviceaccount.Ensure()` in this function, and `serviceaccount.Ensure()` should handle the secret (perhaps by calling `EnsureSecretFromSA()`, of course, but that should be transparent to the caller).

Got it. I missed the context of comment for `Ensure` and only addressed for `EnsureFromYaml`.